### PR TITLE
feat(validator): add backtick escaping validation

### LINKED ISSUE #39: feat(validator): add backtick escaping validation for git/GitHub operations
## Summary

Add validation to detect unescaped backticks in content passed to git commits, PR creation, and GitHub issue creation commands.

## Problem

When using double-quoted strings (`"..."`) with backticks in shell commands, the backticks trigger command substitution, which can cause:

- Silent command execution
- Broken commit messages/PR bodies
- Unexpected behavior

Examples of problematic commands:

```bash
git commit -m "Fix bug in `parser` module"  # backticks interpreted as command substitution
gh pr create --body "Updated `config.toml` handling"  # same issue
gh issue create --body "The `validate` function fails"  # same issue
```

### Option 1: Warn and suggest file-based approach (recommended)

Detect backticks in double-quoted content and suggest using HEREDOC or file-based input:

```bash
# HEREDOC approach
git commit -m "$(cat <<'EOF'
Fix bug in `parser` module
EOF
)"

# File-based approach
gh pr create --body-file pr-body.md
gh issue create --body-file issue-body.md
```

### Option 2: Warn about escaping

Suggest escaping backticks when double quotes are used:

```bash
git commit -m "Fix bug in \`parser\` module"
```

## Scope

Commands to validate:

- `git commit -m "..."` / `git commit --message="..."`
- `gh pr create --body "..."` / `--title "..."`
- `gh issue create --body "..."` / `--title "..."`

## Implementation Notes

- Extend bash parser to detect quote context (single vs double)
- Add `BacktickValidator` in `internal/validators/git/` or new `internal/validators/shell/`
- Consider severity: `Warn` (allow but notify) vs `Fail` (block)
- File-based approach is generally safer and more readable for multiline content

## Acceptance Criteria

- [ ] Detect backticks inside double-quoted strings in relevant commands
- [ ] Provide clear error message with fix suggestions
- [ ] Support both escaping and file-based alternatives in suggestions
- [ ] Add configuration option to enable/disable this validation
- [ ] Unit tests covering various edge cases

### PR DESCRIPTION
**Pull request by bartsmykla**
## Motivation

Claude Code users need validation to detect when backticks in double-quoted strings may cause unexpected command substitution in git commit messages and GitHub operations.

## Summary

Adds BacktickValidator to detect command substitution in double-quoted strings for git commit, gh pr create, and gh issue create commands.

## Implementation information

- **Parser Enhancement** (`pkg/parser/bash.go`, `pkg/parser/command.go`)

  - Added `FindDoubleQuotedBackticks()` method
  - Detects `CmdSubst` nodes within `DblQuoted` parts using AST analysis
  - 11 test cases covering various scenarios

- **BacktickValidator** (`internal/validators/shell/backtick.go`)

  - Validates git commit, gh pr create, gh issue create commands
  - Error code SHELL001 with automatic fix hints
  - Provides 3 fix suggestions: HEREDOC, file-based input, escape backticks
  - 15 test cases

- **Configuration** (`pkg/config/shell.go`)

  - Added `ShellConfig` with `BacktickValidatorConfig`
  - Integrated into root config with nil handling

- **Factory Integration** (`internal/config/factory/shell_factory.go`)

  - Created `ShellValidatorFactory`
  - Predicate-based matching for relevant commands

- **Error Reference System**

  - Added `RefShellBackticks` (SHELL001)
  - Added suggestion to registry
  - Updated documentation

Resolves #39

### CODE REVIEW

File: internal/validators/shell/backtick.go [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+	// argIndex includes command name, so adjust for args slice
+	// If argIndex is 3 for "git commit -m MSG", then:
+	// - 0=git, 1=commit, 2=-m, 3=MSG
+	// In args: 0=commit, 1=-m, 2=MSG
+	// So we need argIndex-1 in the args slice
+	if argIndex <= 1 { // Skip "git" and subcommand
+		return false
+	}
+
+	argsIdx := argIndex - 1 // Adjust for command name
```
[the remark this answers is not in the document]
Comment (bartsmykla) [reply internal/validators/shell/backtick.go]: Fixed in 190fcc3 - extracted the conversion logic into `convertArgIndexToCommandArgsIndex()` helper function with clear documentation about the two index systems.

File: pkg/parser/bash_test.go [hunk 2]
```diff
@@ -359,4 +359,118 @@ EOF`
...
+	Describe("FindDoubleQuotedBackticks", func() {
+		Context("with backticks in double quotes", func() {
+			It("detects backticks in git commit message", func() {
+				issues, err := p.FindDoubleQuotedBackticks(
+					"git commit -m \"Fix bug in `parser` module\"",
+				)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(issues).To(HaveLen(1))
+				Expect(issues[0].ArgIndex).To(Equal(3))
+				// ArgValue will have command substitution executed/removed
```
[the remark this answers is not in the document]
Comment (bartsmykla) [reply pkg/parser/bash_test.go]: Fixed in 190fcc3 - clarified that the parser detects backticks but doesn't execute them. Updated comment to be more accurate.
